### DISCUSSION TITLE
(Bayesian Rule List Classifier) ValueError: All numeric features must be discretized prior to fitting!

### DISCUSSION THREAD

**Discussion by usmanjoy** [csinva/imodels#141] [Q&A]
I transformed my data into discrete value by using EqualWidthDiscretiser from feature engine library
![1](https://user-images.githubusercontent.com/111187501/201969971-462998cc-5f7e-4f04-ac08-4ca96fc72b5a.png)
![2](https://user-images.githubusercontent.com/111187501/201970209-28e4eb5b-9920-4e1c-b04d-cc925e6f10c9.png)
![3](https://user-images.githubusercontent.com/111187501/201971025-ce425f41-c116-40df-b572-0643c2565be9.png)
But when I pass train data into bayesian rule list classifier it always give me value error. what am I doing wrong?
![4](https://user-images.githubusercontent.com/111187501/201971333-0afe7ec2-6351-4e35-ada6-4a5c4471ef7f.png)

**Comment by mfrederico** [comment]
I'm no expert, however if I look at your line 137 its expecting 1 or 0.  

- Have you printed an output of what is in your matrix to be CERTAIN that your x values are 1 OR 0 and that no 'null' value has appeared?
- Is it checking against "registered_days" - that column has > 1 values?
